### PR TITLE
45952398: PackageDeployment/RuntimeManager API Spec

### PR DESCRIPTION
**Pull request by DrusTheAxe**
45952398: PackageDeployment/RuntimeManager API Spec

Package Management APIs

### CODE REVIEW

File: specs/packagemanager/PackageManagement.md [hunk 1]
```diff
@@ -0,0 +1,1136 @@
...
+  - [4.6. PackageRuntimeManager.AddPackageSet()](#46-packageruntimemanageraddpackageset)
+  - [4.7. PackageRuntimeManager.RemovePackageset()](#47-packageruntimemanagerremovepackageset)
+  - [4.6. PackageVolume.Repair()](#46-packagevolumerepair)
+- [5. Remarks](#5-remarks)
+  - [5.1. Platform Support](#51-platform-support)
+- [6. API Details](#6-api-details)
+
+# 2. Background
+
+Windows sports the ability to deploy and manage software as MSIX packages via WinRT APIs in the
```
Comment (cwruss) [inline specs/packagemanager/PackageManagement.md:36]: nit: sports > supports

File: specs/packagemanager/PackageManagement.md [hunk 2]
```diff
@@ -0,0 +1,1136 @@
...
+* Repair a package
+* Reset a package
+* Provision a package
+* Deprovision a package
+
+Additional functionality includes:
+
+* IsReady -- Is a package ready for use?
+* EnsureIsReady -- Is a package ready for use and, if not, make it so
+* IsPackageRegistrationPending -- Is there an update waiting to install?
```
Comment (cwruss) [inline specs/packagemanager/PackageManagement.md:67]: nit: waiting to register?

File: specs/packagemanager/PackageManagement.md [hunk 3]
```diff
@@ -0,0 +1,1136 @@
...
+* Path
+* PackageFamilyName
+* PackageFullName
+* Uri
+
+**NOTE:** Methods with a target of a URI are named ...ByUri....
+
+**NOTE:** Methods with a target of a PackageFamilyName are named ...ByPackageFamilyName....
+
+**NOTE:** Methods with a target of a PackageFamilyName are named ...ByPackageFullName....
```
Comment (cwruss) [inline specs/packagemanager/PackageManagement.md:104]: PackageFullName

File: specs/packagemanager/PackageManagement.md [hunk 4]
```diff
@@ -0,0 +1,1136 @@
...
+|Register     | WAS  |  OS/WAS  |       OS/WAS      |      OS/WAS     | OS/WAS |    X     | OS/WAS  |    WAS     |
+|Remove       |  X   |    X     |        WAS        |      OS/WAS     |   X    |    X     | OS/WAS  |    WAS     |
+|Repair       |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Reset        |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Provision    |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+|Deprovision  |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+
+Legend:
+
+* OS = Supported by Windows (OS) APIs in the Windows.Management.Deployment.PackageManager namespace.
```
Comment (cwruss) [inline specs/packagemanager/PackageManagement.md:130]: By the latest OS version?

File: specs/packagemanager/PackageManagement.md [hunk 5]
```diff
@@ -0,0 +1,1136 @@
...
+**NOTE:** Methods with no ...By<Type>... qualifier accept a string which can contain one or more types
+of identifiers. These can vary for different verbs. See the per-method documentation for the
+specific target types supported by each method.
+
+These methods accept options as a matching `<verb>Package[Set]Options` type, e.g.
+`AddPackageAsync(string packageUri, AddPackageOptions options)`.
+
+The following table shows the supported permutations of verbs and targets:
+
+|Verb         | Path | Filename | PackageFamilyName | PackageFullName | file:  | http(s): | ms-uup: | PackageSet |
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:115]: What's the difference between "path", "filename" and "file:"?
In OS PackageManager APIs these are interchangeable.

File: specs/packagemanager/PackageManagement.md [hunk 6]
```diff
@@ -0,0 +1,1136 @@
...
+
+Thus `EnsurePackageIsReady(pkg, options)` is functionally equivalent to
+
+```c#
+var pdm = PackageDeploymentManager().GetDefault();
+if (!pdm.IsPackageReady(pkg))
+{
+    var result = await pdm.AddPackageAsync(pkg, options);
+}
+endif
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:162]: is "endif" needed?

File: specs/packagemanager/PackageManagement.md [hunk 7]
```diff
@@ -0,0 +1,1136 @@
...
+```
+
+### 3.3.1. Why Is*Ready() Given Ensure*IsReady()?
+
+Ensure*IsReady() performs an 'is ready' check and returns if all is ready. There's no efficiency
+reasons to call Is*Ready() before Ensure*IsReady() (in fact, it's less efficient as Is*Ready() would
+occur twice).
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:169]: Formatting issue: the * characters make everything italic in between.

File: specs/packagemanager/PackageManagement.md [hunk 8]
```diff
@@ -0,0 +1,1136 @@
...
+        var result = await pdm.EnsurePackageIsReadyAsync(pkg, options);
+    }
+}
+```
+
+## 3.4. Repair
+
+`PackageDeploymentManager` offers Repair APIs providing the same functionality as available
+interactively via Settings' `Repair` button on the detail page for an app (via Apps > Installed Apps
+> ... > Advanced options).
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:191]: Formatting issue: this line ends up as a blockquote

File: specs/packagemanager/PackageManagement.md [hunk 9]
```diff
@@ -0,0 +1,1136 @@
...
+
+## 3.5. Reset
+
+`PackageDeploymentManager` offers Reset APIs providing the same functionality as available
+interactively via Settings' `Reset` button on the detail page for an app (via Apps > Installed Apps
+> ... > Advanced options).
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:197]: Formatting issue: this line ends up as a blockquote

File: specs/packagemanager/PackageManagement.md [hunk 10]
```diff
@@ -0,0 +1,1136 @@
...
+}
+return new PackageDeploymentResult(PackageDeploymentStatus.CompletedSuccess);
+```
+
+## 3.8. PackageRuntimeManager
+
+The `PackageRuntimeManager` API provides Dynamic Dependency support for PackageSet operations,
+especially when the caller may not know the exact package(s) involved (for instance, when targeting
+packages via `ms-uup:`). `PackageRuntimeManager` determines the packages involved for a `PackageSet`
+and dynamically adds them caller's package graph.
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:235]: missing word between "them" and "caller's"

File: specs/packagemanager/PackageManagement.md [hunk 11]
```diff
@@ -0,0 +1,1136 @@
...
+
+## 3.9. PackageVolume Repair
+
+Packages are typically<sup>1</sup> installed to a
+[PackageVolume](https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume),
+e.g. `C:\Program Files\WindowsApps` is the default `PackageVolume` on a new Windows system. Windows
+pairs the PackageVolume with the underlying storage volume's media ID to identify the PackageVolume
+even when mount points change (e.g. instead USB flash key and it's mount point is E:, the remove the
+USB flash key and later reinsert it but not it's mounted as K:).
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:244]: Lots of typos in these two lines above

File: specs/packagemanager/PackageManagement.md [hunk 12]
```diff
@@ -0,0 +1,1136 @@
...
+
+It's possible this tracking information can be invalidated, e.g. backup a drive's content, replace
+the drive with a new one and then restore the content. The packages installed on this drive's
+PackageVolume(s) aren't recognized by Windows because the new drive has a different media ID.
+
+The Windows App SDK's new `PackageVolume.Repair()` detects and corrects these conditions.
+
+`PackageVolume.VerifyIfOk()` checks if the PackageVolume is OK or in need of repair.
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:252]: Should this be VerifyIsOk (is, instead of if)?

File: specs/packagemanager/PackageManagement.md [hunk 13]
```diff
@@ -0,0 +1,1136 @@
...
+```c#
+var Install()
+{
+    var packageSet = new PackageSet();
+    var muffin = new PackageSetItem();
+    muffin.PackageUri = new Uri("c:\\contoso\\muffin-1.2.3.4.msix");
+    packageSet.Add(muffin);
+    var waffle = new PackageSetItem();
+    waffle.PackageUri = new Uri("https://contoso.com/waffle-2.4.6.8.msix");
+    packageSet.Add(waffle);
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:364]: How would this work with dependencies?  E.g. if I have package A which depends on frameworks X and Y, and package B which depends on frameworks Y and Z, does this get specified as a set {A, B} with dependencies {X, Y, Z}?  Or is the dependency info part of each PackageSetItem?

File: specs/packagemanager/PackageManagement.md [hunk 14]
```diff
@@ -0,0 +1,1136 @@
...
+    catch (Exception ex)
+    {
+        Console.WriteLine(ex.ToString());
+    }
+}
+```
+
+**NOTE:** This differs from the AddPackageAsync() example by the method name
+(`AddPackageByUriAsync()` instead of `AddPackageAsync()`) and the target type (https: source as URI
+instead of filename as string).
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:349]: Why is the string version needed?  One can just as easily do
            var package = new Uri("d:\\contoso\\muffin.msix");
instead of
            var package = "d:\\contoso\\muffin.msix";

File: specs/packagemanager/PackageManagement.md [hunk 15]
```diff
@@ -0,0 +1,1136 @@
...
+|IsReady      |  X   |    X     |       OS/WAS      |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|EnsureIsReady|  X   |    X     |         X         |        X        |   X    |    X     |  WAS    |    WAS     |
+|Add          | WAS  |    X     |         X         |        X        | OS/WAS |  OS/WAS  | OS/WAS  |    WAS     |
+|Stage        | WAS  |    X     |         X         |        X        | OS/WAS |  OS/WAS  | OS/WAS  |    WAS     |
+|Register     | WAS  |  OS/WAS  |       OS/WAS      |      OS/WAS     | OS/WAS |    X     | OS/WAS  |    WAS     |
+|Remove       |  X   |    X     |        WAS        |      OS/WAS     |   X    |    X     | OS/WAS  |    WAS     |
+|Repair       |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Reset        |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Provision    |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+|Deprovision  |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:126]: In each row, if it accepts PackageSet, is it required that all items in the PackageSet be of the same type?  Is it required that all items in the PackageSet be of types that are allowed on that row (e.g. Provision accepts only PFaN and ms-uup items in its PackageSet, but not Path or PFuN)?

File: specs/packagemanager/PackageManagement.md [hunk 16]
```diff
@@ -0,0 +1,1136 @@
...
+    var muffin = new PackageSetItem();
+    muffin.PackageFamilyName = "contoso.muffin_1234567890abc";
+    muffin.MinVersion = ToVersion(1, 2, 3, 4);
+    muffin.PackageUri = new Uri("c:\\contoso\\muffin-1.2.3.4.msix");
+    packageSet.Add(muffin);
+    var waffle = new PackageSetItem();
+    waffle.PackageFamilyName = "contoso.waffle_1234567890abc";
+    waffle.MinVersion = ToVersion(2, 4, 6, 8);
+    waffle.PackageUri = new Uri("https://contoso.com/waffle-2.4.6.8.msix");
+    packageSet.Add(waffle);
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:408]: How would one know which fields of a PackageSetItem need to be populated for each operation?  Which ones are required, optional, or prohibited?

File: specs/packagemanager/PackageManagement.md [hunk 17]
```diff
@@ -0,0 +1,1136 @@
...
+    }
+    try
+    {
+        packageVolume.Repair();
+    }
+    catch (Exception ex)
+    {
+        Console.WriteLine(ex.ToString());
+    }
+}
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:680]: try-catch block should be inside "if" clause?  If volume is OK then there's no need to repair.

File: specs/packagemanager/PackageManagement.md [hunk 18]
```diff
@@ -0,0 +1,1136 @@
...
+        Boolean IsRegistered{ get; };
+        Guid ActivityId { get; };
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageSetItem
+    {
+        PackageSetItem();
+
+        String Id;
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:815]: How will this be used?

File: specs/packagemanager/PackageManagement.md [hunk 19]
```diff
@@ -0,0 +1,1136 @@
...
+
+        // Package criteria to identify if a matching package is ready for use and to use at runtime
+        String PackageFamilyName;   //required
```
Comment (letao-msft) [inline specs/packagemanager/PackageManagement.md:818]: These comments are specific to using PackageSet in packageRuntimeManager.AddPackageSet or packageRuntimeManager.RemovePackageSet.  They are not true when using PackageSet for other operations like AddPackageSetAsync.

File: specs/packagemanager/PackageManagement.md (hunk 1 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:36]: Was intentional not typo, but OK

File: specs/packagemanager/PackageManagement.md (hunk 5 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:115]: file: is a URI 
path is a directory
file is a filename

**Examples**

* URI vs file = file://c:/foo/bar.msix vs c:\foo\bar.msix
* path vs file = c:\blah vs c:\blah\appxmanifest.xml

OS supports `Register(file://c:/blah/appxmanifest.xml)`. WAS will also support `Register(file://c:/blah)`.

Add and Stage have some similar permutations e.g. `Add(c:\blah, options.StageInPlace=true)`.

Table was wrong, Add+Stage support Path today. Fixed. Checking on Register

File: specs/packagemanager/PackageManagement.md (hunk 15 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:126]: > In each row, if it accepts PackageSet, is it required that all items in the PackageSet be of the same type?

No. A PackageSet is essentially an array of package items (PackageSetItem[], in concept) and each PackageSet has its own data. There's no need for them to have any identical properties; you absolute can have packageset.PackageSetItems[0].PackageUri=file://blah/appxmanifest.xml, [1]=file://blah, [2]=ms-uup://Product/stuff.x64 and [3]=MyFamily_1234567890abc

File: specs/packagemanager/PackageManagement.md (hunk 4 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:130]: That's implied. I didn't want to get into that level of detail; microsoft.com docs do it quite well.

See 5.1. Platform Support for more details about functionality and Windows versions.

File: specs/packagemanager/PackageManagement.md (hunk 6 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:162]: Doh! Thanks

File: specs/packagemanager/PackageManagement.md (hunk 7 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:169]: Ewww. Ahhhh markdown. Thanks. Fixing...

File: specs/packagemanager/PackageManagement.md (hunk 8 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:191]: More unintentional markdown fun. Thanks. Fixing

File: specs/packagemanager/PackageManagement.md (hunk 12 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:252]: Changed to `CheckPackageVolume()` and `RepairPackageVolume()` e.g.

```
var packageVolume = PackageVolumeManager.FindPackageVolumeByPath(path);
if (!packageVolume.CheckPackageVolume())
{
    Console.WriteLine("Needs repair");
    packageVolume.RepairPackageVolume();
    Console.WriteLine("Repaired");
}
```

File: specs/packagemanager/PackageManagement.md (hunk 14 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:349]: Not everything can be expressed as a URI

* YES: file:, http(s):, ms-uup, path, filename
* NO: PackageFamilyName, PackageFullName

Plus "it's easy" in C#, not necessarily so easy in all cases (irksome friction in C++ ABI)

File: specs/packagemanager/PackageManagement.md (hunk 13 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:364]: Those properties are part of the operation's options (e.g. AddPackageOptions); they can't vary per `PackageSetItem` today.

I'd considered adding such things as optional override properties in `PackageSetItem` e.g.

```
    var packageSet = new PackageSet();
    var muffin = new PackageSetItem();
    muffin.PackageUri = new Uri("c:\\contoso\\muffin-1.2.3.4.msix");
    packageSet.Add(muffin);
    var waffle = new PackageSetItem();
    waffle.PackageUri = new Uri("https://contoso.com/waffle-2.4.6.8.msix");
    waffle.Stuff = "foo";
    packageSet.Add(waffle);

    var packageDeploymentManager = PackageDeploymentManager.GetDefault();
    var options = new AddPackageOptions();
    options.Stuff = "bar";
    var deploymentResult = await packageDeploymentManager.AddPackageByUriAsync(packageSet, options);
```

so muffin would be processed with Stuff=bar vs waffle is processed with Stuff=foo

But that brings a host of complexity and questions to the table e.g. PackageSet can be used in multiple contexts but not all support the same options (AddPackageOptions vs StagePackageOptions vs...). Should PackageSetItem have all the options as optional properties (packagesetitem.AddPackageOptions, .StagePackageOptions, ...)? PackageSetItem has all the individual properties as optional (packagesetitem.DevelopmentMode=true, ...)?

We have no urgent need for this variability of data, and we can easily extend the API to support it later when the need/usage is more clearly defined/understood.

TL;DR Out of scope today. Maybe in a future enhancement.

File: specs/packagemanager/PackageManagement.md (hunk 16 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:408]: Documentation.

Nothing's prohibited. Propeties are either be used or ignored/NA, depending on the operation.

That's also true for PackageManager e.g. not all properties of AddPackageOptions are relevant for all Add scenarios.

File: specs/packagemanager/PackageManagement.md (hunk 19 above)
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:818]: Comments are holdover from early implementation. Obsolete now so I've removed them

File: specs/packagemanager/PackageManagement.md [hunk 20]
```diff
@@ -0,0 +1,1134 @@
...
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageSet
+    {
+        PackageSet();
+
+        String Id;
+        IVector<PackageSetItem> PackageSetItems { get; };
+    }
```
Comment (DrusTheAxe) [inline specs/packagemanager/PackageManagement.md:831]: Needs additional property

`Uri PackageUri;`

PackageSet is an array of PackageSetItems where each item has PackageFamilyName+MinVersion+PackageUri (plus other related but those are the key items). That's awesome when you have multiple msix e.g.

PackageSet has PackageSetItems[3] = {
    { family-foo, 1.2.3.4, file://c:/foo/blah.msix },
    { family-bar, 2.4.6.8, file://d:/stuff/bar.msix },
    { family-meh, 3.3.3.3, https://whatever.com/meh.msix },
}

But in the case of ms-uup: URIs you have 1 URI corresponding to N packages. Per the current IDL you can do this with redundancy

PackageSet has PackageSetItems[3] = {
    { family-foo, 1.2.3.4, ms-uup://Product/one.product.has.multiple.packages.x64 },
    { family-bar, 2.4.6.8, ms-uup://Product/one.product.has.multiple.packages.x64 },
    { family-meh, 3.3.3.3, ms-uup://Product/one.product.has.multiple.packages.x64 },
}

Adding a Uri property to the PackageSet object we can remove the redundancy e.g. `

PackageSet.PackageSetItems[3] = {
    { family-foo, 1.2.3.4  },
    { family-bar, 2.4.6.8 },
    { family-meh, 3.3.3.3 },
}
and PackageSet.Uri=ms-uup://Product/one.product.has.multiple.packages.x64.

Treat PackageSet.PackageSetItem[n].Uri is an _override_ for PackageSet.Uri -- if PackageDeploymentManager needs a URI it uses PackageSetItem.Uri if defined, else PackageSet.Uri (else fail)

(Came up after spec posted for review else the spec would already have this) 
RECOMMEND: Add `Uri PackageUri;` to PackageSet

File: specs/packagemanager/PackageManagement.md [hunk 21]
```diff
@@ -0,0 +1,1134 @@
...
+
+        PackageVolume TargetVolume;
+        IVector<Windows.Foundation.Uri> DependencyPackageUris { get; };
+        IVector<String> OptionalPackageFamilyNames { get; };
+        IVector<Windows.Foundation.Uri> OptionalPackageUris { get; };
+        IVector<Windows.Foundation.Uri> RelatedPackageUris { get; };
+        Windows.Foundation.Uri ExternalLocationUri;
+        Windows.Management.Deployment.StubPackageOption StubPackageOption;
+        Boolean AllowUnsigned;
+        Boolean DeveloperMode;
```
Comment (oldnewthing) [inline specs/packagemanager/PackageManagement.md:847]: Does this mean "turn on system developer mode when installing this package"?

File: specs/packagemanager/PackageManagement.md [hunk 22]
```diff
@@ -0,0 +1,1134 @@
...
+        IMap<Windows.Foundation.Uri, String> ExpectedDigests{ get; };
+    }
+
+    // Requires Windows >= 10.0.19041.0 (aka 2004 aka 20H1)
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass RemovePackageOptions
+    {
+        RemovePackageOptions();
+
+        Boolean OkIfNotFound;
```
Comment (oldnewthing) [inline specs/packagemanager/PackageManagement.md:917]: I assume `OkayIfNotFound` defaults to true.

Maybe `SucceedIfNotFound`? or flip it `PackageMustExist`?

File: specs/packagemanager/PackageManagement.md [hunk 23]
```diff
@@ -0,0 +1,1134 @@
...
+    {
+        // Get an instance of the manager
+        static PackageDeploymentManager GetDefault();
+
+        //-------------------------------------------------------------
+        // IsReady
+
+        // Return true if the package(s) are present and available for use
+
+        Boolean IsPackageReady(String package);
```
Comment (oldnewthing) [inline specs/packagemanager/PackageManagement.md:953]: This doesn't have a `By`, so I expect it to take a Package` as the parameter, not a string. What is the string? Package Family Name? Package Full Name? Aumid?

File: specs/packagemanager/PackageManagement.md [hunk 24]
```diff
@@ -0,0 +1,1134 @@
...
+        Boolean SupportsHardLinks{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.isfulltrustpackagesupported
+        Boolean IsFullTrustPackageSupported{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.isappxinstallsupported
+        Boolean IsAppxInstallSupported{ get; };
+
+        /// Return true if the package volume is OK, else false if it needs to be repaired.
+        bool CheckPackageVolume();
```
[the remark this answers is not in the document]
Comment (FrankC-msft) [reply specs/packagemanager/PackageManagement.md:740]: Since this is a method (not property) I presume it's doing work when called to perform the 'check'.  There is precedence for GetIs* method naming.  Would we prefer e.g. `GetIsPackageVolumeHealthy` ?

File: specs/packagemanager/PackageManagement.md [hunk 25]
```diff
@@ -0,0 +1,1134 @@
...
+        PackageVolume();
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.issystemvolume
+        Boolean IsSystemVolume{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.mountpoint
+        String MountPoint{ get; };
+
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume.name
+        String Name{ get; };
```
[the remark this answers is not in the document]
Comment (DrusTheAxe) [reply specs/packagemanager/PackageManagement.md:631]: PackageManager uses the term 'name' in code references (e.g. [FindPackageVolumeByName](https://learn.microsoft.com/uwp/api/windows.management.deployment.packagemanager.findpackagevolume)) and docs' prose mentions name is a media id e.g.

volumeName String
The volume media ID, which is a [GUID](https://learn.microsoft.com/en-us/windows/win32/api/guiddef/ns-guiddef-guid) value.
https://learn.microsoft.com/uwp/api/windows.management.deployment.packagemanager.findpackagevolume#parameters

So we'll stick with the same terminology -- Name is the code term and docs will mention "Volume Name == Volume Media ID"

**Review by DrusTheAxe — COMMENTED** [review]
Resolved feedback

**Review by cwruss — APPROVED** [review]
